### PR TITLE
FIX-#6201: align groupby objects signatures with pandas

### LINKED ISSUE #6201: Align groupby objects signatures with pandas

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6201 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/groupby.py [hunk 1]
```diff
@@ -990,6 +1023,9 @@ def boxplot(
                 ax=ax,
                 figsize=figsize,
                 layout=layout,
+                sharex=False,
+                sharey=True,
+                backend=None,
```
Comment (dchigarev) [inline modin/pandas/groupby.py:1028]: ?
```suggestion
                sharex=sharex,
                sharey=sharey,
                backend=backend,
```

**Review by dchigarev — APPROVED** [review]
looks good, left minor comments

File: modin/pandas/groupby.py [hunk 2]
```diff
@@ -707,28 +736,31 @@ def bfill(self, limit=None):
...
-            agg_kwargs=dict(ddof=ddof),
-            numeric_only=True,
+            agg_kwargs=dict(ddof=ddof, engine=engine, engine_kwargs=engine_kwargs),
+            numeric_only=numeric_only,
         )
 
-    def aggregate(self, func=None, *args, **kwargs):
+    def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs):
+        if engine is not None and engine_kwargs is not None:
+            return self._default_to_pandas(
```
[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/pandas/groupby.py:755]: I will do this right now.

File: modin/pandas/groupby.py [hunk 3]
```diff
@@ -278,12 +278,22 @@ def value_counts(
             )
         )
 
-    def mean(self, numeric_only=None):
+    def mean(self, numeric_only=no_default, engine="cython", engine_kwargs=None):
+        if engine not in ("cython", None) and engine_kwargs is not None:
```
Comment (anmyachev) [inline modin/pandas/groupby.py:282]: `cython` is the default value that pandas converts `None` value to. For example, https://pandas.pydata.org/pandas-docs/version/1.5.3/reference/api/pandas.core.groupby.GroupBy.std.html?highlight=std#pandas.core.groupby.GroupBy.std

**Review by dchigarev — APPROVED** [review]

**Pull request merged**